Sources/BraintreeVenmo/BTVenmoClient.swift

```diff
@@ -207,45 +214,44 @@ import BraintreeCore
             apiClient.post("", parameters: graphQLParameters, httpType: .graphQLAPI) { body, _, error in
                 if let error = error as? NSError {
                     if error.code == BTCoreConstants.networkConnectionLostCode {
-                        self.apiClient.sendAnalyticsEvent("ios.pay-with-venmo.network-connection.failure")
+                        self.apiClient.sendAnalyticsEvent(BTVenmoAnalytics.tokenizeNetworkConnectionLost)
                     }
 
-                    self.apiClient.sendAnalyticsEvent("ios.pay-with-venmo.appswitch.handle.client-failure")
+                    self.apiClient.sendAnalyticsEvent(BTVenmoAnalytics.tokenizeFailed)
                     self.appSwitchCompletion(nil, error)
                     return
                 }
 
                 guard let body else {
+                    self.apiClient.sendAnalyticsEvent(BTVenmoAnalytics.tokenizeFailed)
                     self.appSwitchCompletion(nil, BTVenmoError.invalidBodyReturned)
                     return
                 }
 
                 let venmoAccountNonce: BTVenmoAccountNonce = BTVenmoAccountNonce(with: body)
-                self.apiClient.sendAnalyticsEvent("ios.pay-with-venmo.appswitch.handle.success")
 
                 if self.shouldVault && self.apiClient.clientToken != nil {
                     self.vault(venmoAccountNonce.nonce)
                 } else {
+                    self.apiClient.sendAnalyticsEvent(BTVenmoAnalytics.tokenizeSucceeded)
                     self.appSwitchCompletion(venmoAccountNonce, nil)
                     return
                 }
             }
 
         case .succeeded:
             guard let nonce = returnURL.nonce else {
-                apiClient.sendAnalyticsEvent("ios.pay-with-venmo.appswitch.handle.client-failure")
+                apiClient.sendAnalyticsEvent(BTVenmoAnalytics.tokenizeFailed)
                 appSwitchCompletion(nil, BTVenmoError.invalidReturnURL("nonce"))
                 return
             }
 
             guard let username = returnURL.username else {
-                apiClient.sendAnalyticsEvent("ios.pay-with-venmo.appswitch.handle.client-failure")
+                apiClient.sendAnalyticsEvent(BTVenmoAnalytics.tokenizeFailed)
                 appSwitchCompletion(nil, BTVenmoError.invalidReturnURL("username"))
```

Comment:
I don't want to beat a dead horse - but I still prefer the original idea of using a single "notify" function for each of the 3 end cases (success, fail, cancel). It's annoying to need to duplicate the line `apiClient.sendAnalyticsEvent(BTVenmoAnalytics.tokenizeFailed)` everywhere and is more prone to error.

[We do this in PPCP.](https://github.com/paypal/iOS-SDK/blob/2a79160e81d8757d9c1891ad4e3dc1bf98f64187/Sources/PayPalNativePayments/PayPalNativeCheckoutClient.swift#L103)

Reply:
@KunJeongPark & @sshropshire thoughts?

Reply:
I am open to either. I thought we had a conversation to move away from that to make it clear where the messages were coming from but I may be wrong. I don't have a personal preference with this.

Reply:
I do like the PPCP way. I think it's good our culture allows these discussions to make sure we're well aligned 🙏

Reply:
So other analytic tickets we have finished so far - ApplePay, Card, PayPal, PayPalNativeCheckout, we've taken them out of the helper function to do it this way. Is this suggestion for all future change or start change with this ticket?

Reply:
> So other analytic tickets we have finished so far - ApplePay, Card, PayPal, PayPalNativeCheckout, we've taken them out of the helper function to do it this way. Is this suggestion for all future change or start change with this ticket?

It's up to us, really. You can finish the rest of the modules with the same format you're already doing, and then address the syntax all at once across the SDK. Or each subsequent module you pick up could be using the helper-method style. I would go with the former just to get things out the door, and then make tickets for refactoring with the helper methods. Also feel free to 2nd pass on it at stand-up if you want to get the whole teams opinion.